web/web.go

```diff
@@ -0,0 +1,771 @@
+	var tables []string
+	for t := range app.conv.SpSchema {
+		tables = append(tables, t)
+	}
+	sort.Strings(tables)
+	ddl := make(map[string]string)
+	for _, t := range tables {
+		ddl[t] = app.conv.SpSchema[t].PrintCreateTable(c)
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(ddl)
+}
+
+func getSummary(w http.ResponseWriter, r *http.Request) {
+	reports := internal.AnalyzeTables(app.conv, nil)
+	summary := make(map[string]string)
+	for _, t := range reports {
+		var body string
+		for _, x := range t.Body {
+			body = body + x.Heading + "\n"
+			for i, l := range x.Lines {
+				body = body + fmt.Sprintf("%d) %s.\n", i+1, l) + "\n"
+			}
+		}
+		summary[t.SrcTable] = body
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(summary)
+}
+func writeHeading(s string) string {
+	return strings.Join([]string{
+		"----------------------------\n",
+		s, "\n",
+		"----------------------------\n"}, "")
+}
+func writeStmtStats(driverName string, conv *internal.Conv) string {
+	stmtstats := ""
+	type stat struct {
+		statement string
+		count     int64
+	}
+	var l []stat
+	for s, x := range conv.Stats.Statement {
+		l = append(l, stat{s, x.Schema + x.Data + x.Skip + x.Error})
+	}
+	// Sort by alphabetical order of statements.
+	sort.Slice(l, func(i, j int) bool {
+		return l[i].statement < l[j].statement
+	})
+	stmtstats = stmtstats + writeHeading("Statements Processed")
+	stmtstats = stmtstats + "Analysis of statements in " + driverName + " output, broken down by statement type.\n"
+	stmtstats = stmtstats + "  schema: statements successfully processed for Spanner schema information.\n"
+	stmtstats = stmtstats + "    data: statements successfully processed for data.\n"
+	stmtstats = stmtstats + "    skip: statements not relevant for Spanner schema or data.\n"
+	stmtstats = stmtstats + "   error: statements that could not be processed.\n"
+	stmtstats = stmtstats + "  --------------------------------------\n"
+	stmtstats = stmtstats + fmt.Sprintf("  %6s %6s %6s %6s  %s\n", "schema", "data", "skip", "error", "statement")
+	stmtstats = stmtstats + "  --------------------------------------\n"
+	for _, x := range l {
+		s := conv.Stats.Statement[x.statement]
+		stmtstats = stmtstats + fmt.Sprintf("  %6d %6d %6d %6d  %s\n", s.Schema, s.Data, s.Skip, s.Error, x.statement)
+	}
+	if driverName == "pg_dump" {
+		stmtstats = stmtstats + "See github.com/lfittl/pg_query_go/nodes for definitions of statement types\n"
+		stmtstats = stmtstats + "(lfittl/pg_query_go is the library we use for parsing pg_dump output).\n"
+		stmtstats = stmtstats + "\n"
+	} else if driverName == "mysqldump" {
+		stmtstats = stmtstats + "See https://github.com/pingcap/parser for definitions of statement types\n"
+		stmtstats = stmtstats + "(pingcap/parser is the library we use for parsing mysqldump output).\n"
+		stmtstats = stmtstats + "\n"
+	}
+	return stmtstats
+}
+func getOverview(w http.ResponseWriter, r *http.Request) {
```

Comment:
Could this be replaced by a call to internal.GenerateReport -- this looks like it's doing everything in GenerateReport except the table by table listing and reporting of unexpected at the end?
We could add a parameter to control this, and avoid much of this code.
See the previous comments for handling the bufio.Writer issue.

Reply:
Thanks for the change -- I think that's the right call here.